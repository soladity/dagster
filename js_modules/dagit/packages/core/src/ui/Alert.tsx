import {Colors} from '@blueprintjs/core';
import * as React from 'react';
import styled from 'styled-components/macro';

import {Box} from './Box';
import {ColorsWIP} from './Colors';
import {Group} from './Group';
import {IconName, IconWIP} from './Icon';

export interface Props {
  intent: 'info' | 'warning' | 'error' | 'success';
  title: React.ReactNode;
  description?: React.ReactNode;
}

export const Alert: React.FC<Props> = (props) => {
  const {intent, title, description} = props;

  const {backgroundColor, borderColor, icon, iconColor} = React.useMemo(() => {
    switch (intent) {
      case 'warning':
        return {
          backgroundColor: ColorsWIP.Yellow50,
          borderColor: ColorsWIP.Yellow500,
          icon: 'warning',
          iconColor: ColorsWIP.Yellow500,
        };
      case 'error':
        return {
          backgroundColor: ColorsWIP.Red50,
          borderColor: ColorsWIP.Red500,
          icon: 'error',
          iconColor: ColorsWIP.Red500,
        };
      case 'success':
        return {
          backgroundColor: ColorsWIP.Green50,
          borderColor: ColorsWIP.Green500,
          icon: 'done',
          iconColor: ColorsWIP.Green500,
        };
      case 'info':
      default:
        return {
          backgroundColor: ColorsWIP.Gray50,
          borderColor: ColorsWIP.Blue500,
          icon: 'info',
          iconColor: ColorsWIP.Blue500,
        };
    }
  }, [intent]);

  return (
    <Box
      background={backgroundColor}
      padding={{horizontal: 16, vertical: 12}}
      style={{
        boxShadow: `inset 4px 0 ${borderColor}, inset -1px 1px ${Colors.LIGHT_GRAY3}, inset 0 -1px ${Colors.LIGHT_GRAY3}`,
      }}
    >
      <Group direction="row" spacing={12} alignItems="flex-start">
        <IconWIP name={icon as IconName} color={iconColor} />
        <Group direction="column" spacing={8}>
          <AlertTitle>{title}</AlertTitle>
          {description ? <AlertDescription>{description}</AlertDescription> : null}
        </Group>
      </Group>
    </Box>
  );
};

Alert.defaultProps = {
  intent: 'info',
};

const AlertTitle = styled.div`
  color: ${Colors.DARK_GRAY3};
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
`;

const AlertDescription = styled.div`
  color: ${Colors.DARK_GRAY3};
  font-weight: 400;
`;
