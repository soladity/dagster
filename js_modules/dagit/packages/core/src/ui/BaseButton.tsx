import * as React from 'react';
import styled from 'styled-components/macro';

import {ColorsWIP} from './Colors';
import {IconWrapper} from './Icon';
import {SpinnerWrapper} from './Spinner';
import {FontFamily} from './styles';

interface Props extends React.ComponentPropsWithRef<'button'> {
  icon?: React.ReactNode;
  label?: React.ReactNode;
  loading?: boolean;
  rightIcon?: React.ReactNode;
  fillColor?: string;
  strokeColor?: string;
  textColor?: string;
}

export const BaseButton = React.forwardRef(
  (props: Props, ref: React.ForwardedRef<HTMLButtonElement>) => {
    const {
      fillColor = ColorsWIP.White,
      disabled,
      icon,
      label,
      loading,
      rightIcon,
      textColor = ColorsWIP.Dark,
      strokeColor = ColorsWIP.Gray300,
      ...rest
    } = props;

    return (
      <StyledButton
        {...rest}
        disabled={disabled || loading}
        $fillColor={fillColor}
        $strokeColor={strokeColor}
        $textColor={textColor}
        ref={ref}
      >
        {icon || null}
        {label ? <span>{label}</span> : null}
        {rightIcon || null}
      </StyledButton>
    );
  },
);

interface StyledButtonProps {
  $fillColor: string;
  $strokeColor: string;
  $textColor: string;
}

const StyledButton = styled.button<StyledButtonProps>`
  align-items: center;
  background-color: ${({$fillColor}) => $fillColor || 'transparent'};
  border: none;
  border-radius: 8px;
  color: ${({$textColor}) => $textColor};
  cursor: pointer;
  display: inline-flex;
  flex-direction: row;
  font-family: ${FontFamily.default};
  font-size: 14px;
  line-height: 20px;
  padding: 6px 12px;
  transition: background 100ms, box-shadow 150ms, filter 100ms, opacity 150ms;
  user-select: none;
  box-shadow: ${({$strokeColor}) => `${$strokeColor} inset 0px 0px 0px 1px`};

  :hover {
    box-shadow: ${({$strokeColor}) =>
      `${$strokeColor} inset 0px 0px 0px 1px, rgba(0, 0, 0, 0.12) 0px 2px 12px 0px;`};
  }

  :focus {
    box-shadow: rgba(58, 151, 212, 0.6) 0 0 0 3px;
    outline: none;
  }

  :focus:not(:focus-visible) {
    box-shadow: ${({$strokeColor}) =>
      `${$strokeColor} inset 0px 0px 0px 1px, rgba(0, 0, 0, 0.12) 0px 2px 12px 0px;`};
  }

  :active {
    filter: brightness(0.95);
  }

  :disabled {
    cursor: default;
    opacity: 0.5;
  }

  :disabled:hover {
    box-shadow: ${({$strokeColor}) => `${$strokeColor} inset 0px 0px 0px 1px`};
  }

  ${SpinnerWrapper},
  ${IconWrapper} {
    color: ${({$textColor}) => $textColor};
    background-color: ${({$textColor}) => $textColor};
    align-self: center;
    display: block;
  }

  ${SpinnerWrapper}:first-child,
  ${IconWrapper}:first-child {
    margin-left: -4px;
    margin-right: 4px;
  }

  ${SpinnerWrapper}:last-child,
  ${IconWrapper}:last-child {
    margin-right: -4px;
    margin-left: 4px;
  }

  ${SpinnerWrapper}:first-child:last-child {
    margin: 2px -4px;
  }
  ${IconWrapper}:first-child:last-child {
    margin: 2px -4px;
  }
`;
