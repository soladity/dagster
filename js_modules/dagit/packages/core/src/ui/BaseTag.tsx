import * as React from 'react';
import styled from 'styled-components/macro';

import {ColorsWIP} from './Colors';
import {IconWrapper} from './Icon';

interface Props {
  fillColor?: string;
  textColor?: string;
  icon?: React.ReactNode;
  rightIcon?: React.ReactNode;
  label?: React.ReactNode;
}

export const BaseTag = (props: Props) => {
  const {
    fillColor = ColorsWIP.Gray10,
    textColor = ColorsWIP.Gray900,
    icon,
    rightIcon,
    label,
  } = props;
  return (
    <StyledTag $fillColor={fillColor} $textColor={textColor}>
      {icon || null}
      {label ? <span>{label}</span> : null}
      {rightIcon || null}
    </StyledTag>
  );
};

interface StyledTagProps {
  $fillColor: string;
  $textColor: string;
}

const StyledTag = styled.div<StyledTagProps>`
  background-color: ${({$fillColor}) => $fillColor};
  border-radius: 8px;
  color: ${({$textColor}) => $textColor};
  cursor: default;
  display: inline-flex;
  flex-direction: row;
  font-size: 12px;
  line-height: 16px;
  align-items: center;
  padding: 4px 8px;
  user-select: none;

  > ${IconWrapper}:first-child {
    margin-right: 4px;
    margin-left: -4px;
  }

  > ${IconWrapper}:last-child {
    margin-left: 4px;
    margin-right: -4px;
  }

  > ${IconWrapper}:first-child:last-child {
    margin: 0 -4px;
  }
`;
