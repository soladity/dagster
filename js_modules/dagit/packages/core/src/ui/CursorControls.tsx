import {Button, Icon} from '@blueprintjs/core';
import {IconNames} from '@blueprintjs/icons';
import * as React from 'react';
import styled from 'styled-components/macro';

export interface CursorPaginationProps {
  hasPrevCursor: boolean;
  hasNextCursor: boolean;
  popCursor: () => void;
  advanceCursor: () => void;
  reset: () => void;
}

export const CursorPaginationControls: React.FunctionComponent<CursorPaginationProps> = ({
  hasPrevCursor,
  hasNextCursor,
  popCursor,
  advanceCursor,
}) => {
  return (
    <div style={{textAlign: 'center', marginBottom: 10}}>
      <Button
        style={{
          marginRight: 4,
        }}
        disabled={!hasPrevCursor}
        icon={IconNames.ARROW_LEFT}
        onClick={popCursor}
      >
        Prev Page
      </Button>
      <Button
        style={{
          marginLeft: 4,
        }}
        disabled={!hasNextCursor}
        rightIcon={IconNames.ARROW_RIGHT}
        onClick={advanceCursor}
      >
        Next Page
      </Button>
    </div>
  );
};

export const CursorHistoryControls: React.FunctionComponent<CursorPaginationProps> = ({
  hasPrevCursor,
  hasNextCursor,
  popCursor,
  advanceCursor,
}) => {
  return (
    <CursorHistoryControlsContainer>
      <Button style={{marginRight: 4}} disabled={!hasNextCursor} onClick={advanceCursor}>
        <Icon icon={IconNames.ARROW_LEFT} />
        <span className="hideable-button-text" style={{marginLeft: 8}}>
          Older
        </span>
      </Button>
      <Button style={{marginLeft: 4}} disabled={!hasPrevCursor} onClick={popCursor}>
        <span className="hideable-button-text" style={{marginRight: 8}}>
          Newer
        </span>
        <Icon icon={IconNames.ARROW_RIGHT} />
      </Button>
    </CursorHistoryControlsContainer>
  );
};

const CursorHistoryControlsContainer = styled.div`
  text-align: center;
  white-space: nowrap;

  @media (max-width: 1000px) {
    & .hideable-button-text {
      display: none;
    }
  }
`;
