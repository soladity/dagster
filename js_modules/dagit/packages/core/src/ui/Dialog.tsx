import {Dialog as BlueprintDialog} from '@blueprintjs/core';
import * as React from 'react';
import styled, {createGlobalStyle} from 'styled-components/macro';

import {Box} from './Box';
import {ColorsWIP} from './Colors';
import {Group} from './Group';
import {IconName, IconWIP} from './Icon';

interface Props extends Omit<React.ComponentProps<typeof BlueprintDialog>, 'backdropClassName'> {}

export const DialogWIP: React.FC<Props> = (props) => {
  const {...rest} = props;
  return <BlueprintDialog {...rest} backdropClassName="dagit-backdrop" className="dagit-dialog" />;
};

interface HeaderProps {
  icon?: IconName;
  label: React.ReactNode;
}

export const DialogHeader: React.FC<HeaderProps> = (props) => {
  const {icon, label} = props;
  return (
    <Box
      background={ColorsWIP.Gray50}
      padding={{vertical: 16, horizontal: 20}}
      border={{side: 'bottom', width: 1, color: ColorsWIP.Gray200}}
    >
      <Group direction="row" spacing={8} alignItems="center">
        {icon ? <IconWIP name={icon} color={ColorsWIP.Gray800} /> : null}
        <DialogHeaderText>{label}</DialogHeaderText>
      </Group>
    </Box>
  );
};

export const DialogBody: React.FC = (props) => {
  return (
    <Box padding={{vertical: 16, horizontal: 20}} background={ColorsWIP.White}>
      {props.children}
    </Box>
  );
};

interface DialogFooterProps {
  buttons?: React.ReactNode[];
  left?: React.ReactFragment;
}

export const DialogFooter: React.FC<DialogFooterProps> = (props) => {
  return (
    <Box
      padding={{bottom: 20, top: 8, horizontal: 20}}
      background={ColorsWIP.White}
      flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}
    >
      <div>{props.left}</div>
      <Group direction="row" spacing={12} alignItems="center">
        {props.buttons}
      </Group>
    </Box>
  );
};

const DialogHeaderText = styled.div`
  font-size: 16px;
  font-weight: 500;
`;

export const GlobalDialogStyle = createGlobalStyle`
  .bp3-overlay-backdrop {
    background-color: rgba(189, 186, 183, 0.7);
  }

  .bp3-dialog-container {
    display: grid;
    grid-template-rows: minmax(40px, 1fr) auto minmax(40px, 2fr);
    grid-template-columns: 40px 8fr 40px;
  }

  .bp3-dialog {
    background-color: ${ColorsWIP.White};
    border-radius: 4px;
    box-shadow: rgba(0, 0, 0, 0.12) 0px 2px 12px;
    grid-row: 2;
    grid-column: 2;
    margin: 0 auto;
    padding: 0;
  }

  .bp3-dialog > :first-child {
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
  }

  .bp3-dialog > :last-child {
    border-bottom-right-radius: 4px;
    border-bottom-left-radius: 4px;
  }

  .bp3-dialog-container.bp3-overlay-enter > .bp3-dialog,
  .bp3-dialog-container.bp3-overlay-appear > .bp3-dialog {
    opacity: 0;
    transform:scale(0.95);
  }

  .bp3-dialog-container.bp3-overlay-enter-active > .bp3-dialog,
  .bp3-dialog-container.bp3-overlay-appear-active > .bp3-dialog {
    opacity: 1;
    transform: scale(1);
    transition-delay: 0;
    transition-duration: 150ms;
    transition-property: opacity, transform;
    transition-timing-function: ease-in-out;
  }

  .bp3-dialog-container.bp3-overlay-exit > .bp3-dialog {
    opacity: 1;
    transform: scale(1);
  }

  .bp3-dialog-container.bp3-overlay-exit-active > .bp3-dialog{
    opacity: 0;
    transform: scale(0.95);
    transition-delay:0;
    transition-duration: 150ms;
    transition-property: opacity, transform;
    transition-timing-function: ease-in-out;
  }
`;
