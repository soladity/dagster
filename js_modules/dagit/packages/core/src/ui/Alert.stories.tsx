import {Story, Meta} from '@storybook/react/types-6-0';
import * as React from 'react';

import {Alert, Props} from './Alert';
import {ButtonLink} from './ButtonLink';

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Alert',
  component: Alert,
} as Meta;

const Template: Story<Props> = (props) => <Alert {...props} />;

export const Example = Template.bind({});
Example.args = {
  intent: 'info',
  title: 'This pipeline run is queued.',
  description: (
    <div>
      Click <ButtonLink>here</ButtonLink> to proceed.
    </div>
  ),
};
