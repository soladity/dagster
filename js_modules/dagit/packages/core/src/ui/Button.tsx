// eslint-disable-next-line no-restricted-imports
import {Button as BlueprintButton} from '@blueprintjs/core';
import * as React from 'react';

import {BaseButton} from './BaseButton';
import {ColorsWIP} from './Colors';
import {Spinner} from './Spinner';

type BlueprintIntent = React.ComponentProps<typeof BlueprintButton>['intent'];
type BlueprintOutlined = React.ComponentProps<typeof BlueprintButton>['outlined'];

const intentToFillColor = (intent: BlueprintIntent, outlined: BlueprintOutlined) => {
  if (outlined) {
    return 'transparent';
  }
  switch (intent) {
    case 'primary':
      return ColorsWIP.Gray900;
    case 'danger':
      return ColorsWIP.Red500;
    case 'success':
      return ColorsWIP.Green500;
    case 'warning':
      return ColorsWIP.Yellow500;
    case 'none':
    default:
      return 'transparent';
  }
};

const intentToTextColor = (intent: BlueprintIntent, outlined: BlueprintOutlined) => {
  if (outlined) {
    switch (intent) {
      case 'primary':
        return ColorsWIP.Gray900;
      case 'danger':
        return ColorsWIP.Red500;
      case 'success':
        return ColorsWIP.Green500;
      case 'warning':
        return ColorsWIP.Yellow500;
      case 'none':
      default:
        return ColorsWIP.Dark;
    }
  }
  return !intent || intent === 'none' ? ColorsWIP.Dark : ColorsWIP.White;
};

const intentToStrokeColor = (intent: BlueprintIntent, outlined: BlueprintOutlined) => {
  if (!intent || intent === 'none' || outlined) {
    switch (intent) {
      case 'primary':
        return ColorsWIP.Gray900;
      case 'danger':
        return ColorsWIP.Red500;
      case 'success':
        return ColorsWIP.Green500;
      case 'warning':
        return ColorsWIP.Yellow500;
      case 'none':
      default:
        return ColorsWIP.Gray300;
    }
  }
  return 'transparent';
};

const intentToSpinnerColor = (intent: BlueprintIntent, outlined: BlueprintOutlined) => {
  if (outlined) {
    switch (intent) {
      case 'primary':
        return ColorsWIP.Gray600;
      case 'danger':
        return ColorsWIP.Red500;
      case 'success':
        return ColorsWIP.Green500;
      case 'warning':
        return ColorsWIP.Yellow500;
      case 'none':
      default:
        return ColorsWIP.Gray600;
    }
  }
  return !intent || intent === 'none' ? ColorsWIP.Gray600 : ColorsWIP.White;
};

export const ButtonWIP = React.forwardRef(
  (
    props: React.ComponentProps<typeof BlueprintButton>,
    ref: React.ForwardedRef<HTMLButtonElement>,
  ) => {
    const {children, icon, intent, loading, outlined, rightIcon, ...rest} = props;

    const fillColor = intentToFillColor(intent, outlined);
    const textColor = intentToTextColor(intent, outlined);
    const strokeColor = intentToStrokeColor(intent, outlined);

    let iconOrSpinner = icon;
    let rightIconOrSpinner = rightIcon;

    if (loading) {
      const spinnerColor = intentToSpinnerColor(intent, outlined);
      iconOrSpinner = icon ? <Spinner purpose="body-text" fillColor={spinnerColor} /> : icon;
      rightIconOrSpinner =
        rightIcon && !icon ? <Spinner purpose="body-text" fillColor={spinnerColor} /> : rightIcon;
    }

    return (
      <BaseButton
        {...rest}
        icon={iconOrSpinner}
        rightIcon={rightIconOrSpinner}
        loading={loading}
        fillColor={fillColor}
        textColor={textColor}
        strokeColor={strokeColor}
        label={children}
        ref={ref}
      />
    );
  },
);

ButtonWIP.displayName = 'Button';
