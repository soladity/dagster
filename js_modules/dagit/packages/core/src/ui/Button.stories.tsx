import {Meta} from '@storybook/react/types-6-0';
import * as React from 'react';

import {ButtonWIP as Button} from './Button';
import {Group} from './Group';
import {IconWIP as Icon} from './Icon';

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Button',
  component: Button,
} as Meta;

export const Default = () => {
  return (
    <Group direction="column" spacing={8}>
      <Button>Button</Button>
      <Button icon={<Icon name="star" />}>Button</Button>
      <Button rightIcon={<Icon name="close" />}>Button</Button>
      <Button icon={<Icon name="source" />} rightIcon={<Icon name="expand_more" />}>
        Button
      </Button>
      <Button icon={<Icon name="cached" />} />
    </Group>
  );
};

export const Intent = () => {
  return (
    <Group direction="column" spacing={8}>
      <Button icon={<Icon name="star" />}>No intent set</Button>
      <Button icon={<Icon name="star" />} intent="primary">
        Primary
      </Button>
      <Button icon={<Icon name="done" />} intent="success">
        Success
      </Button>
      <Button icon={<Icon name="error" />} intent="danger">
        Danger
      </Button>
      <Button icon={<Icon name="warning" />} intent="warning">
        Warning
      </Button>
      <Button icon={<Icon name="star" />} intent="none">
        None
      </Button>
    </Group>
  );
};

export const Disabled = () => {
  return (
    <Group direction="column" spacing={8}>
      <Button icon={<Icon name="star" />} intent="primary">
        Enabled
      </Button>
      <Button icon={<Icon name="cached" />} disabled intent="primary">
        Disabled
      </Button>
    </Group>
  );
};
