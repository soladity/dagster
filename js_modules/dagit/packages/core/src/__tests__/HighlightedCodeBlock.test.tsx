import * as React from 'react';
import {BrowserRouter} from 'react-router-dom';
import * as TestRenderer from 'react-test-renderer';

import {HighlightedCodeBlock} from '../ui/HighlightedCodeBlock';

it('correctly renders keywords inside a comment', () => {
  const component = TestRenderer.create(
    <BrowserRouter>
      <HighlightedCodeBlock
        language="sql"
        value={
          '-- this comment contains a dash - do not highlight if drop table\n' +
          'select * from q2_on_time_data\n' +
          "where origin = 'SFO'"
        }
        style={{
          height: 510,
          margin: 0,
          overflow: 'scroll',
          fontSize: '0.9em',
        }}
      />
    </BrowserRouter>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
