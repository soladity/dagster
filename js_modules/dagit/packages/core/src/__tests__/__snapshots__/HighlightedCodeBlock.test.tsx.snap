// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders keywords inside a comment 1`] = `
<pre
  style={
    Object {
      "backgroundColor": "transparent",
      "margin": 0,
      "padding": 0,
    }
  }
>
  -- this comment contains a dash - do not highlight if drop table
select * from q2_on_time_data
where origin = 'SFO'
</pre>
`;
